pkg/models/appconfiguration/workload/workload.go

```diff
@@ -22,7 +22,7 @@ type Workload struct {
 	*Job     `yaml:",inline" json:",inline"`
 }
 
-func (w *Workload) MarshalJSON() ([]byte, error) {
```

Comment:
why do you replace the reference with instance? All structs containing `Workload` are used as a reference.

Reply:
Test comment

Reply:
Using `(w *Workload)` cannot serialize `Workload{}` instance as expected, only serialize `&Workload{}`.
However, using `(w Workload)` can serialize `Workload{}` and `&Workload{}` instance as expected.
You can refer to the `MarshalJSON` and `UnmarshalJSON` implementation of `Time` in the standard library:
https://sourcegraph.com/github.com/golang/go/-/blob/src/time/time.go?L1360:15

Reply:
ok, i got it